fw/fe310/test/Makefile

```diff
@@ -16,15 +16,14 @@ app_:
 	(cd app && $(MAKE)) || exit;
 
 %.o: %.c
-	riscv32-none-elf-gcc $(CFLAGS) -c $<
+	$(CC) $(CFLAGS) -c $<
 
 $(TARGET): app_ $(DEPS)
-	riscv32-none-elf-gcc $(DEPS) $(LDFLAGS) -o $@
+	$(CC) -L$(out) $(DEPS) $(LDFLAGS) -o $@
```

Comment:
Does this work? I guess maybe `$out` is being set by the shellHook/phase? Either way, I think `$out` should be renamed to something more descriptive (perhaps `$NANO_LIBS_PATH`?), documented in the flake and makefile, and then set when running the makefile. E.g. something like `NANO_LIBS_PATH=${some.derivation}/lib make -C fe310` as a sample invocation from a flake.

Reply:
Agreed, some sort of override or wrapper is needed that does a renaming. This might even be worth a PR to nixpkgs if the compiler is expecting `${LIB_NAME}_nano.a`. This is something worth asking the author about IMO

This might not be the cleanest solution (I would ask around) but even just a simple derivation that only has an `installPhase` that does the renaming would be clearer, and save on duplicate code. You could then set more descriptive environment variable in the fe310 derivation's invocation to the makefile. You could (similarly) set the environment variable in exactly the same way in the shellHook.